docs: added guide on updating status-bar color based on theme (#5438)
unoplatform/uno: doc/articles/features/status-bar-theme-color.md

> # Updating StatusBar color based on dark/light theme

Reviewer: ```suggestion
# How to update StatusBar color based on dark/light theme
```

> The [`UISettings.ColorValuesChanged` event](https://docs.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.uisettings.colorvalueschanged?view=winrt-19041) can be used to listen when the dark mode is turned on or off.

Reviewer: ```suggestion
The [`UISettings.ColorValuesChanged` event](https://docs.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.uisettings.colorvalueschanged?view=winrt-19041) can be used to listen for notifications when dark mode is enabled or disabled at the system level.
```

## Example
The complete sample code can be found here: [StatusBarThemeColor](https://github.com/unoplatform/Uno.Samples/tree/master/UI/StatusBarThemeColor)

> Or, you can follow along the step-by-step instructions:

Reviewer: ```suggestion

## Step-by-step instructions
```

> 1. Create a new uno app: [get started](..\get-started.md)

Reviewer: ```suggestion
1. Create a new Uno Platform application, following the instructions [here](..\get-started.md).
```

2. In `MainPage.xaml`, add a `<CommandBar>`:

>     > On iOS, the status bar color cannot be set directly, so it is done via through a CommandBar placed in the page. You could also use any xaml element like `<Grid>` or `<Border>` to achieve similar effect, if your application doesnt use any navigation or doesn't use native navigation. This is because the page content can goes under the status bar. In fact, you usually have to add padding to avoid that (see next step).

Reviewer: ```suggestion
    > On iOS, the status bar color cannot be set directly, so it is done via through a `CommandBar` placed in the page. You could also use any XAML element like `<Grid>` or `<Border>` to achieve a similar effect, if your application doesn't use navigation or doesn't use native navigation. This is because the page content can go under the status bar. In fact, you usually have to add padding to avoid that (see next step).
```

    ```xml
    <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
        <Grid.RowDefinitions>
            <RowDefinition Height="Auto" />
            <RowDefinition Height="*" />
        </Grid.RowDefinitions>

        <CommandBar Grid.Row="0"
                    x:Name="MyCommandBar"
                    Content="MainPage" />

        <StackPanel Grid.Row="1">
            <TextBlock Text="Hello, world!"
                        Margin="20"
                        FontSize="30" />
        </StackPanel>
    </Grid>
    ```

> 3. Add a `VisibleBoundsPadding.PaddingMask` the root `<Grid>` to prevent the content being stuck under status & command bar:

Reviewer: ```suggestion
3. Add a [`VisibleBoundsPadding.PaddingMask`](../features/VisibleBoundsPadding.md) to the root `<Grid>` to prevent the content being partially covered by the status and command bars:
```

    ```xml
    <Page ...
          xmlns:toolkit="using:Uno.UI.Toolkit"
          ...>

        <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}"
              toolkit:VisibleBoundsPadding.PaddingMask="Top">
    ```

> 4. In `MainPage.xaml.cs`, expose the `MyCommandBar` which will be referenced in later step:

Reviewer: ```suggestion
4. In `MainPage.xaml.cs`, expose the `MyCommandBar` which will be referenced in a later step:
```

    ```cs
    public static MainPage Instance { get; private set; }

    public MainPage()
    {
        Instance = this;

        this.InitializeComponent();
    }

    public CommandBar GetCommandBar() => MyCommandBar;
    ```

5. In `App.xaml.cs`, enable the native frame navigation to use the native style for `CommandBar`, instead of the UWP style:
    ```cs
    public App()
    {
        ConfigureFilters(global::Uno.Extensions.LogExtensionPoint.AmbientLoggerFactory);

    #if __IOS__ || __ANDROID__
        // Enable native frame navigation.
        Uno.UI.FeatureConfiguration.Style.ConfigureNativeFrameNavigation();
    #endif

        this.InitializeComponent();
        this.Suspending += OnSuspending;
    }
    ```

6. Subscribe to the `UISettings.ColorValuesChanged` event from `App.xaml.cs`:

>     > Note that the instance of `UISettings` is kept to prevent it from disposed when going out of scope, which will dispose the event subscription.

Reviewer: ```suggestion
    > Note that the instance of `UISettings` is kept to prevent it from disposed when going out of scope, which would otherwise dispose the event subscription.
```

        // note: For ios in dark mode, setting this value will have no effect.
        var foreground = isDarkMode ? Windows.UI.Colors.White : Windows.UI.Colors.Black;
        Windows.UI.ViewManagement.StatusBar.GetForCurrentView().ForegroundColor = foreground;

        // === 3. Set the background color.
        var background = isDarkMode ? Windows.UI.Colors.MidnightBlue : Windows.UI.Colors.SkyBlue;
    #if __ANDROID__
        // On Android, this is done by calling Window.SetStatusBarColor.
        if (Uno.UI.ContextHelper.Current is Android.App.Activity activity)
        {
            activity.Window.SetStatusBarColor(background);
        }
    #endif
        // On iOS, this is done via the native CommandBar which goes under the status bar.
        // For android, we will also update the CommandBar just for consistency.
        if (MainPage.Instance.GetCommandBar() is CommandBar commandBar)
        {
            commandBar.Foreground = new SolidColorBrush(foreground); // controls the color for the "MainPage" page title
            commandBar.Background = new SolidColorBrush(background);
        }
    #endif
    }

## Concluding remarks

> This is just one way to show how it could be done. Alternatively, depending on the application, the follow options can also be considered:

Reviewer: ```suggestion
This is just one way to show how it could be done. Alternatively, depending on the application, the following options can also be considered:
```

> - Attached Property on the `CommandBar` (when multiple pages are used)

Reviewer: ```suggestion
- [Attached Property](https://docs.microsoft.com/en-us/windows/uwp/xaml-platform/attached-properties-overview) on the `CommandBar` (when multiple pages are used)
```
